actors/market/src/lib.rs

```diff
@@ -719,17 +738,20 @@ impl Actor {
                         }
 
                         // Delete the proposal (but not state, which doesn't exist).
-                        let deleted = st.remove_proposal(rt.store(), deal_id)?;
-
-                        if deleted.is_none() {
-                            return Err(actor_error!(
-                                illegal_state,
-                                format!(
-                                    "failed to delete deal {} proposal {}: does not exist",
-                                    deal_id, dcid
-                                )
-                            ));
-                        }
+                        let _deleted =
+                            st.remove_proposal(rt.store(), deal_id)?.ok_or_else(|| {
```

Comment:
Revert this one, it's an internal state error.

Reply:
Done 👍🏾